src/System.Runtime.Extensions/tests/System/IO/PathTests.netcoreapp.cs

```diff
@@ -172,5 +172,177 @@ public static void GetFileNameWithSpaces_Unix_Span()
+            AssertExtensions.Throws<ArgumentException>("basePath", () => Path.GetFullPath("", @"foo\bar"));
+            AssertExtensions.Throws<ArgumentException>("basePath", () => Path.GetFullPath("tmp", @"foo\bar"));
+            AssertExtensions.Throws<ArgumentException>("basePath", () => Path.GetFullPath(@"C:\git", @"foo\bar"));
+            AssertExtensions.Throws<ArgumentNullException>("path", () => Path.GetFullPath(null, @"foo\bar"));
+			Assert.Throws<ArgumentNullException>("path", () => Path.GetFullPath(null, @"C:\foo\bar"));
+			Assert.Throws<ArgumentException>(() => Path.GetFullPath("C:\\gi\0t", @"c:\foo\bar"));
+		}
+
+		[Fact]
+		[PlatformSpecific(TestPlatforms.AnyUnix)]
+		public static void GetFullPath_BasePath_InvalidInput_Unix()
+		{
+			AssertExtensions.Throws<ArgumentNullException>("basePath", () => Path.GetFullPath("", null));
+			AssertExtensions.Throws<ArgumentNullException>("basePath", () => Path.GetFullPath("tmp", null));
+			AssertExtensions.Throws<ArgumentNullException>("basePath", () => Path.GetFullPath(@"\home", null));
+			AssertExtensions.Throws<ArgumentException>("basePath", () => Path.GetFullPath("", @"foo\bar"));
+			AssertExtensions.Throws<ArgumentException>("basePath", () => Path.GetFullPath("tmp", @"foo\bar"));
+            AssertExtensions.Throws<ArgumentException>("basePath", () => Path.GetFullPath(@"\home", @"foo\bar"));
+            AssertExtensions.Throws<ArgumentNullException>("path", () => Path.GetFullPath(null, @"foo/bar"));
+            AssertExtensions.Throws<ArgumentNullException>("path", () => Path.GetFullPath(null, @"/foo/bar"));
+			Assert.Throws<ArgumentException>(() => Path.GetFullPath("/gi\0t", @"/foo/bar"));
+		}
+
+		public static IEnumerable<object[]> GetFullPath_BasePath_BasicExpansions_TestData_Windows()
+		{
+			string curDir = @"c:\git\corefx";
+			yield return new object[] { curDir, curDir, curDir };
+			yield return new object[] { "", curDir, curDir };
+			yield return new object[] { "..", curDir, Path.GetDirectoryName(curDir) };
+			yield return new object[] { Path.Combine(curDir, ".", ".", ".", ".", "."), curDir, curDir };
+			yield return new object[] { curDir + new string(Path.DirectorySeparatorChar, 3) + ".", curDir, curDir };
+			yield return new object[] { Path.Combine(curDir, "..", Path.GetFileName(curDir), ".", "..", Path.GetFileName(curDir)), curDir, curDir };
+			yield return new object[] { Path.Combine(Path.GetPathRoot(curDir), "somedir", ".."), curDir, Path.GetPathRoot(curDir) };
+			yield return new object[] { Path.Combine(Path.GetPathRoot(curDir), "."), curDir, Path.GetPathRoot(curDir) };
+			yield return new object[] { Path.Combine(Path.GetPathRoot(curDir), "..", "..", "..", ".."), curDir, Path.GetPathRoot(curDir) };
+			yield return new object[] { Path.GetPathRoot(curDir) + new string(Path.DirectorySeparatorChar, 3), curDir, Path.GetPathRoot(curDir) };
+
+			// Device paths
+			yield return new object[] { "foo", @"\\?\C:\ ", @"\\?\C:\ \foo" };
```

Comment:
The core scenarios are the path having relative segments, not the base path. You should add cases with relative segments on the path.